src/flamenco/runtime/tests/fd_exec_instr_test.c

```diff
@@ -1883,8 +1911,61 @@ int
 __wrap_fd_execute_instr( fd_exec_txn_ctx_t * txn_ctx,
                          fd_instr_info_t *   instr_info )
 {
-    (void)(txn_ctx);
-    (void)(instr_info);
-    FD_LOG_WARNING(( "fd_execute_instr is disabled" ));
+    static const pb_byte_t zero_blk[32] = {0};
+
+    if( cpi_exec_effects == NULL ) {
+      FD_LOG_WARNING(( "fd_execute_instr is disabled" ));
+      return FD_EXECUTOR_INSTR_SUCCESS;
+    }
+
+    // Iterate through instruction accounts
+    for( ushort i = 0UL; i < instr_info->acct_cnt; ++i ) {
+      uchar idx_in_txn = instr_info->acct_txn_idxs[i];
+      fd_pubkey_t * acct_pubkey = &instr_info->acct_pubkeys[i];
+
+      fd_borrowed_account_t * acct = NULL;
+      /* Find (first) account in cpi_exec_effects->modified_accounts that matches pubkey */
+      for( uint j = 0UL; j < cpi_exec_effects->modified_accounts_count; ++j ) {
+        fd_exec_test_acct_state_t * acct_state = &cpi_exec_effects->modified_accounts[j];
+        if( memcmp( acct_state->address, acct_pubkey, sizeof(fd_pubkey_t) ) != 0 ) continue;
+
+        /* Fetch borrowed account */
+        /* First check if account is read-only. 
+           TODO: have fd_txn_borrowed_account_modify_idx perform this check? */
+
+        if( fd_txn_borrowed_account_view_idx( txn_ctx, idx_in_txn, &acct ) ) {
+          break;
+        }
+        if( acct->meta == NULL ){
```

Comment:
Add a comment explaining that we can remove this restriction when direct mapping is enabled, as this invariant will be enforced by the address translation routine.